Common/src/Common/ProtoSerializer.cs

```diff
@@ -15,73 +15,48 @@
 {
 // *** you need some mechanism to map types to fields
-  private static readonly IDictionary<int, Type> typeLookup = new List<Type>
-                                                              {
-                                                                typeof(ProtoNative<int>),
-                                                                typeof(int),
-                                                                typeof(uint),
-                                                                typeof(long),
-                                                                typeof(ulong),
-                                                                typeof(double),
-                                                                typeof(float),
-                                                                typeof(short),
-                                                                typeof(byte),
-                                                                typeof(string),
-                                                                typeof(int[]),
-                                                                typeof(uint[]),
-                                                                typeof(long[]),
-                                                                typeof(ulong[]),
-                                                                typeof(double[]),
-                                                                typeof(float[]),
-                                                                typeof(short[]),
-                                                                typeof(byte[]),
-                                                                typeof(string[]),
-                                                                typeof(Nullable),
-                                                                typeof(ProtoArray),
-                                                                typeof(IEnumerable),
-                                                                typeof(IDictionary),
-                                                                typeof(Array),
-                                                                typeof(ArmonikPayload),
-                                                              }.Select((t,
-                                                                        idx) => new
-                                                                                {
-                                                                                  idx,
-                                                                                  t,
-                                                                                })
-                                                               .ToDictionary(x => x.idx,
-                                                                             x => x.t);
-
-  public byte[] SerializeMessageObjectArray(object[] values)
```

Comment:
Serialization and deserialization for message object arrays have been replaced by the generic version.